### PR TITLE
Search plugin fixes

### PR DESCRIPTION
**Pull request by shaneriley**
Fixes:
* https://webarchive.jira.com/browse/WEBDEV-3838
* https://webarchive.jira.com/browse/WEBDEV-3839
* https://webarchive.jira.com/browse/WEBDEV-3840
* https://webarchive.jira.com/browse/WEBDEV-3841
* https://webarchive.jira.com/browse/WEBDEV-3844

### CODE REVIEW

File: src/js/plugins/search/plugin.search.js [hunk 1]
```diff
@@ -333,7 +333,7 @@ BookReader.prototype.updateSearchHilites2UP = function() {
  */
 BookReader.prototype.removeSearchHilites = function() {
   const results = this.searchResults;
-  if (null == results) return;
+  if (null == results || !results.matches) { return; }
```
Comment (iisa) [inline src/js/plugins/search/plugin.search.js:336]: thanks

File: src/js/plugins/search/view.js [hunk 2]
```diff
@@ -169,22 +166,48 @@ class SearchView {
...
+    }
     this.br.$('.BRnavline .BRsearch').eq(++this.currentMatchIndex).click();
     this.updateResultsPosition();
+    this.updateSearchNavigationButtons();
+  }
+
+  getClosestMatchIndex(comparisonFn) {
+    const matchPages = this.matches.map((m) => m.par[0].page);
+    const currentPage = this.br.currentIndex() + 1;
+    const closestTo = (pool, comparator) => {
```
Comment (iisa) [inline src/js/plugins/search/view.js:191]: this is cool! can we add a mini doc on its goodness?

**Review by iisa — APPROVED** [review]
looks good! solid clean up sweep, too. 🍰

File: src/js/plugins/search/view.js (hunk 2 above)
Comment (shaneriley) [reply src/js/plugins/search/view.js:191]: You mean explaining the binary search algorithm? I could add something in there.

**Pull request merged**